department-of-veterans-affairs/vets-website
# Certificate of Eligibility | Restore city & state after unchecking military base

## Original Tickets

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/42216
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/44009

## URL of change

`/housing-assistance/home-loans/request-coe-form-26-1880/mailing-address`

## Background

An issue was found in form 526 and 686C-674  where the Veteran would submit the form with their city set to `APO`, which is an abbreviation for "Air/Army Post Office", indicating that the Veteran lives on a military base outside of the United States; but the reviewer noted that the Veteran(s) had addresses within the United States. A bug was found in the logic of form 526 & 686C-674 address ui schemas and has been fixed. Note that the code for these forms was copied from the platform `profileAddress` definition and modified previously.

The Certificate of Eligibility form, not yet in production, also has this issue. But the address ui schema it uses is from platform's `profileAddress`. While updating the logic to restore the city and state after unchecking the military base checkbox, it seemed easier to also include the ["Standardize platform UI schema" work](https://github.com/department-of-veterans-affairs/va.gov-team/issues/44009) so that the duplicate code in 526 and 686C-674 can eventually be removed.

## Steps to test

1. Pull this branch or test this code in the review instance 
2. Log into user 228 (Colder)
3. Navigated to `/housing-assistance/home-loans/request-coe-form-26-1880`
4. Fill out the form and get to the URL of change above
5. Enter a U.S. City & State
6. Check the "I live on a United States military base outside the U.S." checkbox
7. Select a military city & state
8. Now uncheck the "I live on a..." checkbox
9. The previous city & state will be restored
10. Re-checking the "I live on a..." checkbox will clear out the military city & state selects and show an error (since the fields had been previously touched)
11. Unchecking the "I live on a..." checkbox will again restore the U.S. City & State entered in step 5

## Code changes

Updated:
- Platform forms-systems profile address definition
- Added a function (`$`) as a shortcut for `querySelector`
- Added `profileAddress` unit tests - which were missing
- Added a `mock-form` example using this updated profile address definition. Note, these changes require adding `updateFormData` to the `config/form.js` page.
- Updated COE address code to work with the updated profile address definition

## How was your code tested

Added unit tests

## Acceptance criteria
- [x] Standardize profile address keys
- [x] Updated COE to fix APO address issue
- [x] Updated mock-form example

## Definition of done
